spec/lib/claim_letters/utils/user_helper_spec.rb

```diff
@@ -0,0 +1,476 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'claim_letters/utils/user_helper'
+
+RSpec.describe ClaimLetters::Utils::UserHelper do
+  describe '.safe_get' do
+    context 'when user is a Hash' do
+      context 'with symbol keys' do
+        let(:user) { { file_number: '123456', ssn: '987654321', participant_id: 'P123' } }
+
+        it 'retrieves value using symbol key' do
+          expect(described_class.safe_get(user, :file_number)).to eq('123456')
+        end
+
+        it 'retrieves value when attribute is passed as string' do
+          expect(described_class.safe_get(user, 'file_number')).to eq('123456')
+        end
+
+        it 'returns nil for non-existent key' do
+          expect(described_class.safe_get(user, :non_existent)).to be_nil
+        end
+      end
+
+      context 'with string keys' do
+        let(:user) { { 'file_number' => '123456', 'ssn' => '987654321', 'participant_id' => 'P123' } }
+
+        it 'retrieves value using string key' do
+          expect(described_class.safe_get(user, 'file_number')).to eq('123456')
+        end
+
+        it 'retrieves value when attribute is passed as symbol' do
+          expect(described_class.safe_get(user, :file_number)).to eq('123456')
+        end
+
+        it 'returns nil for non-existent key' do
+          expect(described_class.safe_get(user, :non_existent)).to be_nil
+        end
+      end
+
+      context 'with mixed keys' do
+        let(:user) { { file_number: '123456', 'ssn' => '987654321' } }
+
+        it 'retrieves symbol key with symbol attribute' do
+          expect(described_class.safe_get(user, :file_number)).to eq('123456')
+        end
+
+        it 'retrieves string key with string attribute' do
+          expect(described_class.safe_get(user, 'ssn')).to eq('987654321')
+        end
+      end
+
+      context 'with nil or blank values' do
+        let(:user) { { file_number: nil, ssn: '', participant_id: '   ' } }
+
+        it 'returns nil for nil value' do
+          expect(described_class.safe_get(user, :file_number)).to be_nil
+        end
+
+        it 'returns empty string for empty string value' do
+          expect(described_class.safe_get(user, :ssn)).to eq('')
+        end
+
+        it 'returns whitespace string as-is' do
+          expect(described_class.safe_get(user, :participant_id)).to eq('   ')
+        end
+      end
+    end
+
+    context 'when user is an Object' do
+      let(:user_class) do
```

Comment:
Non-blocking: This user class definition is repeated 5 times in this file. It appears this is the current pattern so makes sense for consistency, but we could clean this up later by extracting it to a support file if we want to reduce duplication in the future.